minidump-common/src/format.rs

```diff
@@ -4,6 +4,26 @@
 //! some [Breakpad][breakpad] and [Crashpad][crashpad] extension types are defined here and should
 //! match the definitions from those projects.
 //!
+//! # Type Layouts
+//!
+//! This library isn't a "proper" minidump-sys library because it doesn't use repr attributes
+//! to force Rust to layout these structs identically to how they're laid out in memory.
+//!
+//! The reasons for this are 3-fold:
+//!
+//! 1. It isn't necessary because we specify how to serialize/deserialize things with `scroll`
+//!    via `derive(Pread, Pwrite)` which uses the declared field order and not the in-memory
+//!    layout, and assumes everything is packed anyway (which as a rule, minidump types are).
```

Comment:
Note, Windows headers use `#pragma pack(4)` for all structures. So they are packed but only on 4 bytes boundaries. In practice I don't think it matters because I've never encountered holes but still, might be worth mentioning.

Reply:
good insight!